src/link_analysis.rs

```diff
@@ -225,3 +225,162 @@ pub fn pagerank(
+/// and convergence is not guaranteed. The function will stop when `max_iter`
+/// iterations is reached or when the computed vector between two iterations
+/// is smaller than the error tolerance multiplied by the number of nodes.
+///
+/// In the case of multigraphs the weights of any parallel edges will be
+/// summed when computing the hubs and authorities.
+///
+/// :param PyDiGraph graph: The graph object to run the algorithm on
+/// :param weight_fn: An optional input callable that will be passed the edge's
+///     payload object and is expected to return a `float` weight for that edge.
+///     If this is not specified 1.0 will be used as the weight
+///     for every edge in ``graph``
+/// :param dict nstart: Optional starting value for the power iteration for each node.
+/// :param float tol: The error tolerance used when checking for convergence in the
+///     power method. If this is not specified default value of 1e-8 is used.
+/// :param int max_iter: The maximum number of iterations in the power method. If
+///     not specified a default value of 100 is used.
+/// :param boolean normalized: If the scores should be normalized (defaults to True).
+///
+/// :returns: a tuple of read-only dict-like object whose keys are the node indices. The first value in the tuple
+///      contain the hubs scores. The second value contains the authority scores.
+/// :rtype: tuple[CentralityMapping, CentralityMapping]
+#[pyfunction(
+    signature = (
+        graph,
+        weight_fn=None,
+        nstart=None,
+        tol=1e-6,
+        max_iter=100,
+        normalized=true,
+    )
+)]
+#[pyo3(
+    text_signature = "(graph, /, weight_fn=None, nstart=None, tol=1.0e-8, max_iter=100, normalized=True)"
+)]
+pub fn hits(
+    py: Python,
+    graph: &PyDiGraph,
+    weight_fn: Option<PyObject>,
+    nstart: Option<HashMap<usize, f64>>,
+    tol: f64,
+    max_iter: usize,
+    normalized: bool,
+) -> PyResult<(CentralityMapping, CentralityMapping)> {
+    // we use the node bound to make the code work if nodes were removed
+    let n = graph.graph.node_count();
+    let mat_size = graph.graph.node_bound();
+    let node_indices: Vec<usize> = graph.graph.node_indices().map(|x| x.index()).collect();
+
+    // Handle empty case
+    if n == 0 {
+        return Ok((
+            CentralityMapping {
+                centralities: DictMap::new(),
+            },
+            CentralityMapping {
+                centralities: DictMap::new(),
+            },
+        ));
+    }
+
+    // Grab the graph weights from Python to Rust
+    let mut adjacent: HashMap<(usize, usize), f64> =
+        HashMap::with_capacity(graph.graph.edge_count());
+    let default_weight: f64 = 1.0;
+
+    for edge in graph.graph.edge_references() {
+        let i = NodeIndexable::to_index(&graph.graph, edge.source());
+        let j = NodeIndexable::to_index(&graph.graph, edge.target());
+        let weight = edge.weight().clone();
```

Comment:
It's marginally more efficient to give pyo3 the ref directly so we don't need to look it up from the current context.

```suggestion
        let weight = edge.weight().clone_ref(py);
```